addons/mrp/models/stock_rule.py

```diff
@@ -118,9 +118,13 @@ def _prepare_mo_vals(self, product_id, product_qty, product_uom, location_id, na
         }
         if location_id.get_warehouse().manufacture_steps == 'pbm_sam':
             # Use the procurement group created in _run_pull mrp override
+            # Preserve the origin from the original stock move, if available
+            if len(values.get('move_dest_ids', [])) == 1 and values['move_dest_ids'].origin:
```

Comment:
are you sure `values['move_dest_ids']` is always a recordset?

Reply:
Good point, normally it comes from https://github.com/odoo/odoo/blob/a8224115d8254b4200416f552fc86f6e9e68223c/addons/stock/models/stock_move.py#L1063, but it's not impossible that it would be overridden to return an array instead.  I changed the code to use explicit indexing.